blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/JSONMetric.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2013 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.rackspacecloud.blueflood.inputs.formats;
+
+import com.rackspacecloud.blueflood.service.Configuration;
+import com.rackspacecloud.blueflood.service.CoreConfig;
+
+// Jackson compatible class. Jackson uses reflection to call these methods and so they have to match JSON keys.
+public class JSONMetric {
+
+    public static final long BEFORE_CURRENT_COLLECTIONTIME_MS = Configuration.getInstance().getLongProperty( CoreConfig.BEFORE_CURRENT_COLLECTIONTIME_MS );
+    public static final long AFTER_CURRENT_COLLECTIONTIME_MS = Configuration.getInstance().getLongProperty( CoreConfig.AFTER_CURRENT_COLLECTIONTIME_MS );
+
+    private String metricName;
+    private Object metricValue;
+    private long collectionTime;
+    private int ttlInSeconds;
+    private String unit;
+
+    public String getMetricName() {
+        return metricName;
+    }
+
+    public void setMetricName(String metricName) {
+        this.metricName = metricName;
+    }
+
+    public String getUnit() {
+        return unit;
+    }
+
+    public void setUnit(String unit) {
+        this.unit = unit;
+    }
+
+    public Object getMetricValue() {
+        return metricValue;
+    }
+
+    public void setMetricValue(Object metricValue) {
+        this.metricValue = metricValue;
+    }
+
+    public long getCollectionTime() {
+        return collectionTime;
+    }
+
+    public void setCollectionTime(long collectionTime) {
+        this.collectionTime = collectionTime;
+    }
+
+    public int getTtlInSeconds() {
+        return this.ttlInSeconds;
+    }
+
+    public void setTtlInSeconds(int ttlInSeconds) {
+        this.ttlInSeconds = ttlInSeconds;
+    }
+
+    public String validate() {
+        String error = "";
+
+        long currentTime = System.currentTimeMillis();
+        if ( tooFuture(currentTime, collectionTime) ) {
+            error = "'" + metricName + "': 'collectionTime' '" + collectionTime + "' is more than '" + AFTER_CURRENT_COLLECTIONTIME_MS + "' milliseconds into the future.";
+        } else if ( tooPast(currentTime, collectionTime) ) {
+            error = "'" + metricName + "': 'collectionTime' '" + collectionTime + "' is more than '" + BEFORE_CURRENT_COLLECTIONTIME_MS + "' milliseconds into the past.";
+        }
+
+        return error;
+    }
+
+    public static boolean tooPast(long currentTime, long collectionTime) {
```

Comment:
I really hate duplicating logic.  The classes structure are different, but our idea of what metrics (for either type regular or aggregated) are too far in the past or too far in the future for ingestion is the same.  That's why I originally thought of either:

1)  put the validation code in its own class that can be share with any metrics type (regular or aggregated or otherwise),

2)  create a base class with common validation and properties methods but extend it to radically different metrics (which I tried but didn't like so much so I scrapped it), or

So we can do those two or just go back to option

3) have the tooPast/tooFuture methods which will have the exact validation logics in both AggregratedPayload and JSONMetric class.  

The most correct me is 1) a common validation class, or 2)  have a common base class for both, and maybe rename AggreggatedPayload to be JSONMetricAggregated or something.  

For rebasing, the main thing is that at the moment, we have a JSONMetricContainer that parses and hold json metrics and mapped metric, which is used by HttpMetricsIngestionHandler, but we don't have one for AggregatedPayload class (the same logics is just thrown into the handle() method of HttpAggregatedMultiIngestionHandler class).

But give me another +1 on duplicating the logic of tooPast, tooFuture and I'll do it.